### PR TITLE
Feature/add more log subscribers

### PR DESCRIPTION
**Pull request by MarcGrimme**
### TL;DR

This PR includes the following changes:
1. Bump ruby version to 2.1.4
2. Introduce new log subscribers for active_record and action_view
3. Introduce ``request_context`` for being used with rake and delayed jobs
4. Not necessarily need the ``ActionController`` monkey patch but directly include the new method to the controller in the Rails App 

### Problem

Logstasher currently only supports the instrumentation done from ``ActionController#process_action``. Nevertheless the other instrumentations specially the ones from ActiveRecord are very important to be passed to external logging too. This should be addressed here.

Plus the problem that Rails if run in other contexts like as cron job aka ``rake`` or as part of ``delayed_job`` is not using the valuable concept of the ``request_context``. This should also be addressed in that PR.

Last but not least it is IMHO very dangerous to implicitly patch the ``ActionController#process_action``. As a second configurable approach there is a switch where the monkey patch can be disabled and directly be imported into the controllers in the Rails app like
```ruby
class ApplicationController < ActionController::Base
  ..
  if Module.const_defined?('LogStasher') and LogStasher.enabled
    include LogStasher::ActionController::Instrumentation
  end
  ..
```
this. This makes it visible to the developer that something is included in that controller.

Still Works-In-Progress as tests are missing for the new log-subscribers.

### CODE REVIEW

**Comment by shadabahmed** [comment]
Hi. This is a huge commit. would you like to rebase this on the current master since I've merged multiple PRs today

**Comment by MarcGrimme** [comment]
Rebase done. Let me know what you think.

**Comment by shadabahmed** [comment]
Thanks man :+1:

**Comment by shadabahmed** [comment]
`Pushed logstasher 0.8.0 to rubygems.org`

**Comment by ctaintor** [comment]
@MarcGrimme - could you add some of the features to the README?
